lib/Controller/PageController.php

```diff
@@ -92,6 +100,22 @@ public function index(): TemplateResponse
             ]
         );
 
+        $usersettings = [
+            'compact',
+            'compactExpand',
+            'preventReadOnScroll',
+            'oldestFirst',
+            'showAll'
+        ];
+        foreach ($usersettings as $setting) {
+            $this->initialState->provideInitialState($setting, $this->settings->getUserValue(
+                $this->getUserId(),
+                $this->appName,
+		$setting,
+	        '0')
+            );
+        }
```

Comment:
Do you have some link to documentation for this? Based on the function naming this seems like something that should happen once and not for every API call.

Reply:
https://docs.nextcloud.com/server/latest/developer_manual/basics/front-end/js.html#loading-initial-state
Unfortunately it is nowhere explained where to put the php part, but looking at other apps like notes, deck. The all have it in the TemplateResponse of the PageController.